packages/jest-snapshot/src/State.js

```diff
@@ -83,7 +83,7 @@ class SnapshotState {
 
     const isEmpty = Object.keys(this._snapshotData).length === 0;
 
-    if ((this._dirty || this._uncheckedKeys.size) && !isEmpty) {
+    if ((this._dirty || this._uncheckedKeys.size || update) && !isEmpty) {
```

Comment:
This is added because otherwise `saveSnapshotFile` wouldn't be called (adding a version comment doesn't mark snapshot "dirty")